serialization/src/main/java/com/twitter/serial/stream/bytebuffer/ByteBufferSerializerOutput.java

```diff
@@ -254,70 +257,24 @@ private void ensureCapacity(int sizeNeeded) {
      * Encodes the string into the buffer using UTF-8.
      */
     private void encodeString(@NotNull String string) {
-        final int length = string.length();
-        final int headerPosition = mByteBuffer.position();
-        writeIntHeader(SerializerDefs.TYPE_STRING_ASCII, length);
+        ByteString byteString = ByteString.encodeUtf8(string);
```

Comment:
final

Reply:
Also, you should move this into the else below

Reply:
Also, this is copying the string as well. Why are we removing this code at all? Is this code suspect as well?